tycho-storage/src/postgres/contract.rs

```diff
@@ -1034,68 +1033,27 @@ impl PostgresGateway {
             creation_tx: creation_tx_id,
             created_at: Some(created_ts),
             deleted_at: None,
-            balance: new.native_balance.clone(),
-            code: new.code.clone(),
-            code_hash: new.code_hash.clone(),
         };
         let hex_addr = hex::encode(&new.address);
 
-        let account_id = {
-            use schema::account::dsl;
-            diesel::insert_into(schema::account::table)
-                .values(new_contract.new_account())
-                .on_conflict(on_constraint("account_chain_id_address_key"))
-                .do_update()
-                .set((
-                    dsl::title.eq(excluded(dsl::title)),
-                    dsl::creation_tx.eq(excluded(dsl::creation_tx)),
-                    dsl::created_at.eq(excluded(dsl::created_at)),
-                ))
-                .returning(schema::account::id)
-                .get_result::<i64>(db)
-                .await
-                .map_err(|err| {
-                    error!("Failed inserting account");
-                    storage_error_from_diesel(err, "Account", &hex_addr, None)
-                })?
-        };
+        use schema::account::dsl;
+        diesel::insert_into(schema::account::table)
+            .values(new_contract.new_account())
+            .on_conflict(on_constraint("account_chain_id_address_key"))
+            .do_update()
```

Comment:
Since it is called insert, I think this should error if the contract already exists.